plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -225,12 +232,141 @@ public static function get_json_schema(): array {
 							'intersectionRect'   => $dom_rect_schema,
 							'boundingClientRect' => $dom_rect_schema,
 						),
-						'additionalProperties' => false,
+
+						// Additional properties may be added to the schema for items of elements via the od_url_metric_schema_element_item_additional_properties filter.
+						// See further explanation below.
+						'additionalProperties' => true,
 					),
 				),
 			),
-			'additionalProperties' => false,
+			// Additional root properties may be added to the schema via the od_url_metric_schema_root_additional_properties filter.
+			// Therefore, additionalProperties is set to true so that additional properties defined in the extended schema may persist
+			// in a stored URL Metric even when the extension is deactivated. For REST API requests, the OD_Strict_URL_Metric
+			// which sets this to false so that newly-submitted URL Metrics only ever include the known properties.
+			'additionalProperties' => true,
```

Comment:
The explanation sort of makes sense to me, but where specifically is it relevant to have `additionalProperties` set to `true` for existing URL metrics?

Reply:
It is relevant for when URL metrics are stored, such as when they were collected at a time when the schema was extended so additional properties were allowed. If `additionalProperties` aren't allowed, then those previously-stored URL metrics would be immediately invalid when an extending plugin is deactivated.

Reply:
I get the idea, but my question was meant to be: Where in WordPress's and this plugin's code execution does that actually become relevant? Where would `additionalProperties` be used, if not in the REST API? Like is that a real problem or just a theoretical problem?

Reply:
If `additionalProperties` is `true` it's the same as if `additionalProperties` isn't set at all. Its presence here is just to have a fully documented schema. The `OD_Strict_URL_Metric` subclass extends the `get_json_schema` method to switch `additionalProperties` to `false` and this subclass is what is used with the REST API. The REST API should only accept currently-recognized properties, so that's why the "strict" version is used.

Reply:
Just to share this here: We clarified this outside of the PR, this is relevant because the JSON schema is also used to parse URL metrics from existing posts - so that's the use-case where `additionalProperties` needs to be `true`.